unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -467,11 +469,15 @@ public void Initialize(ServerAction action)
                 actionFinished(false);
                 return;
             }
+
             else
             {
                 gridSize = action.gridSize;
                 StartCoroutine(checkInitializeAgentLocationAction());
             }
+
+            //initialize how long the default wait time for objects to stop moving is
+            this.TimeToWaitForObjectsToComeToRest = action.TimeToWaitForObjectsToComeToRest;
```

Comment:
To auto detect if an object has been moving I thought we had mentioned using it's velocity as the derivative of it's position checking if the difference between the positions between two or more frames was zero. Is this used somewhere else? do we want this extra parameter for wait-time or is it that to get that non-movement it would take too long?

Reply:
it does both. What will happen is the action will return depending on which happens first:
1) action times out because it waited the entire duration of TimeToWait
2) object comes to rest via velocity check, and the action returns regardless of time left

I got a request to limit this to some default time because the default behavior of only waiting until all objects come to rest could result in cases where an action must delay 30+ seconds. This new behavior will still allow this, but it will require users to set the TimeToWait to somethine like a minute themselves.

Reply:
cool